tests/components/bsblan/test_climate.py

```diff
@@ -0,0 +1,299 @@
+"""Tests for the BSB-Lan climate platform."""
+
+from unittest.mock import AsyncMock, PropertyMock, patch
+
+from bsblan import BSBLANError
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.bsblan.climate import BSBLANClimate
+from homeassistant.components.bsblan.const import ATTR_TARGET_TEMPERATURE
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    ATTR_PRESET_MODE,
+    DOMAIN as CLIMATE_DOMAIN,
+    PRESET_ECO,
+    PRESET_NONE,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+import homeassistant.helpers.entity_registry as er
+
+from . import setup_with_selected_platforms
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+ENTITY_ID = "climate.bsb_lan"
+
+
+@pytest.mark.parametrize("static_file", ["static.json"])
+async def test_climate_entity_properties(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    static_file: str,
+) -> None:
+    """Test the initial parameters in Celsius."""
+    await mock_bsblan.set_static_values(static_file)
+    await setup_with_selected_platforms(hass, mock_config_entry, [Platform.CLIMATE])
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+    climate_entity = hass.data[CLIMATE_DOMAIN].get_entity(ENTITY_ID)
+
+    # Test when current_temperature is "---"
+    with patch.object(
+        mock_bsblan.state.return_value.current_temperature, "value", "---"
+    ):
+        await hass.helpers.entity_component.async_update_entity(ENTITY_ID)
+        state = hass.states.get(ENTITY_ID)
```

Comment:
Make the entity update naturally, so instead of doing this (I think this is a polling integration)) use the freezer and skip the same amount of time until the next update. It will then naturally update the entity

Reply:
I'll check the freezer